### ISSUE TITLE
Absolute paths for files within tar archives

### ISSUE THREAD

**Issue by d-Rickyy-b** [mholt/archiver#262]
## What would you like to have changed?

Ability to enable storing of files in tar archive with their absolute paths.

## Why is this feature a useful, necessary, and/or important addition to this project?

You might not want to archive every subdirectory within your crafted structure. So you might specify a more deeply nested directory but want to preserve higher directories (maybe even up to the file system root) for restoring the files later on.

An example:

Imagine you are storing files in /home/rico/project/ and within that directory you have the following structure:
```
project/
├── huge_files
├── images
├── src
└── tmp
```

Now I'd prefer to get an archive file with the contents of `src` and `images` and avoid adding the tmp and "huge" files to my archive, because that's not what I need in this case. On the other hand I still want to preserve the directory structure to know "where it belongs to". Maybe I have several projects all having a "src" folder. Knowing by the archive paths where which archive belongs to is quite helpful. The final archive would contain the following paths & files:

```
/home/rico/project/src/<files>
/home/rico/project/images/<files>
```

## What alternatives are there, or what are you doing in the meantime to work around the lack of this feature?

1. When possible using tar -cvf /absolute/path/to/file
2. Modifying your code to add a new check
-> https://github.com/mholt/archiver/blob/25e050dbb4e3c591402916201098241bb661a2f7/tar.go#L303-L317
I have added one line in line 311: 
```topLevelFolder = filepath.Dir(source)```
That way we always get the full path.

## Please link to any relevant issues, pull requests, or other discussions.

#250

**Comment by coolaj86** [comment]
For the record: I actually agree with you, personally.

But as per the discussion in #250, we probably won't change the behavior.

However, I would like to know why you want to use `arc` rather than `tar` (or Go's native tar package) for this use case?

**Comment by d-Rickyy-b** [comment]
Thanks for your comment. One reason why I want to use archiver is because it provides a **way** better API than the native tar package.
Another reason is, that I also wanted zip support. And archiver abstracts the different archive formats away. One package for multiple formats.

Actually I went with using the tar package for now, because it gives me the freedom to use absolute paths - but I would be way happier with the archiver API.

**Comment by mholt** [comment]
This is fixed in #302, where you specify the mapping of file paths on disk to the file paths in the archive when you create the archive. Please check it out! :) We can reopen this issue if it is still unresolved or needs more discussion.
